### PR TITLE
[wasm] Detect the OS through the browser.

### PR DESCRIPTION
**Pull request by kjpou1**
Resolves issue: https://github.com/mono/mono/issues/13002

### CODE REVIEW

File: sdks/wasm/library_mono.js [hunk 1]
```diff
@@ -358,7 +358,33 @@ var MonoSupportLib = {
...
+	// Win32NT = 2,
+	// WinCE = 3,
+	// Unix = 4,
+	// // Since NET 3.5 SP1 or silverlight
+	// Xbox = 5,
+	// MacOSX = 6,
+	mono_wasm_os_platform_id: function () {
+		if (ENVIRONMENT_IS_WEB) {
+			var platformenv = window.navigator.platform.toUpperCase();
+			if (platformenv.indexOf("MAC")>=0
```
Comment (kg) [inline sdks/wasm/library_mono.js]: This kind of substring match is too fragile, I think you need to have whitespace at the beginning or end of the match patterns.

Comment (vargaz) [reply sdks/wasm/library_mono.js]: Agree, I think we should return WebAssembly or something.

Comment (kjpou1) [reply sdks/wasm/library_mono.js]: We can setup another PlatformId for WebAssembly.

**Comment by kjpou1** [comment]
Issue opened here: https://github.com/dotnet/standard/issues/1073

**Pull request closed without merging**